### PR TITLE
Architecture flowchart

### LINKED ISSUE #52: Make a flow chart
- Start with the [old one](https://cdc-my.sharepoint.com/:wb:/r/personal/ulp7_cdc_gov/Documents/Whiteboards/MCMV%20summer%20project%20chat.whiteboard?d=wa01dcea03fc247e28aaf5d973cf606b2&csf=1&web=1&e=5nUf5A)
- linmod.data is the common data pre-processing
- each "model" has its own pre-processing
- Use this to think about whether it makes sense for Evaluators to be able to query FitModels

### PR DESCRIPTION
**Pull request by swo**
- [x] Add architecture diagram to documents
- [ ] Clarify if this reflects current reality or hoped-for reality
- [ ] Figure out if we should name different components (eg with `linmod.whatever` classes)

Resolves #52 

cc @afmagee42 @thanasibakis

### CODE REVIEW

**Comment by afmagee42** [comment]
We're missing a link from "clean data" to "evaluator."

**Comment by swo** [comment]
> We're missing a link from "clean data" to "evaluator."

good catch, I added eval data as a separate stream, separate from the training data

**Comment by afmagee42** [comment]
> Clarify if this reflects current reality or hoped-for reality

This is _close_ to reality.
- FWIW I don't think we currently have any "common pre-processing." 
- The "evaluator" is currently a few lines of code in `main.py`, which is fine, but the convergence diagnostic stuff is also a few lines in `main.py` so it feels a bit off balance?
- The links to the visualization aren't quite right. We don't visualize the scores, we take the forked clean eval data (good call on separating that out) and combine it directly with the model outputs to make the eval viz.

> Figure out if we should name different components (eg with linmod.whatever classes)

Probably best not to at this point, since the boxes make conceptual sense but don't correspond neatly in the same ways to the same sorts of scripts/classes/functions.

File: docs/architecture.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Architecture
+
+- Raw data comes in whatever form
```
Comment (afmagee42) [inline docs/architecture.md:3]: Re: format data comes in on, how close to reality do we want to hew here, and how aspirational do we want to be? Reality _could_ be we take in anything from which we can get counts of taxa per location/group per day/timestep. Reality is that we read in line-list data directly from one specific source.

File: docs/architecture.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+- They get cleaned up to some common schema
```
Comment (afmagee42) [inline docs/architecture.md:4]: Worth elaborating that our common schema is count per taxon-location-day? (Or more accurately, that augmented with time in "days after the forecast data)

File: docs/architecture.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+- A "model" is everything between the common-schema input and some common-schema output
```
Comment (afmagee42) [inline docs/architecture.md:5]: Worth elaborating on our common output schema? Posterior samples of proportions per taxon-location-day for all days in a range?

File: docs/architecture.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+- There might be further post-processing beyond that, that is still upstream of visualizations and so forth
+- An "evaluator" is some thing that coordinates the action of multiple score functions
```
Comment (afmagee42) [inline docs/architecture.md:7]: How into the count-based evaluation do we want to get here? Evaluators are also in charge of getting, from models, predictions of lineage counts conditioned on total observation sizes from the clean eval data

**Review by afmagee42 — APPROVED** [review]
Some comments/questions/suggestions but nothing so heinous that we couldn't merge it. I think these comments (https://github.com/CDCgov/cfa-viral-lineage-model/pull/54#issuecomment-2407991647) are bigger than any of the in-lines

**Comment by swo** [comment]
I'll move these to an issue, so we can at least get this merged and get started on it

**Pull request merged**